Tests/Indicators/TimeSeriesForecastTests.cs

```diff
@@ -0,0 +1,64 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using NUnit.Framework;
+using QuantConnect.Indicators;
+
+namespace QuantConnect.Tests.Indicators
+{
+[TestFixture]
+public class TimeSeriesForecastTests
```

Comment:
The test fixture should inherit from `CommonIndicatorTests`, like other indicator tests, so it runs the expected test suite

nit - Indentation is off

Reply:
Thanks, I ran the test suite and it passed all unit tests. I have attached the code that generated the`spy_tsf.csv` file below if you would like to take a look:
```py
from datetime import datetime
import tulipy as ti
import pandas as pd
import numpy as np

df = pd.read_csv(
    "https://github.com/QuantConnect/Lean/raw/master/Data/equity/usa/daily/spy.zip",
    index_col=0,
    names=["datetime", "open", "high", "low", "close", "volume"],
)

def date_convert(date):
    return datetime.strptime(date, '%Y%m%d %H:%M').strftime('%m/%d/%Y')

def time_convert(date):
    return datetime.strptime(date, '%Y%m%d %H:%M').strftime('%H:%M')

df = df.reset_index().rename(columns={'index':'datetime'})
df['date'] = df['datetime'].apply(date_convert)
df['time'] = df['datetime'].apply(time_convert)

close = np.array(df.close, dtype=np.float64)

period = 5
tsf = ti.tsf(close, period=period)

# when the size of the data is less than the period, insert zeros at the beginning
tsf = np.insert(tsf, 0, np.zeros(period - 1))
df["tsf"] = tsf
df = df[['date', 'time', 'open', 'high', 'low', 'close', 'volume', 'tsf']]

print(df.head())
df.to_csv("spy_tsf.csv")
```

Reply:
As discussed in https://github.com/QuantConnect/Lean/pull/8155#issuecomment-2217862574

Wondering if we shouldn't have this script in Lean repository (as we have for comparison with talipp
https://github.com/QuantConnect/Lean/blob/a7efcfe9d5182c671385b99666019d399d181121/Tests/TestData/generate_reference_data_from_talipp.py#L20

Any opinion @Martin-Molinero @jaredbroad ?